src/Xamarin.Android.Build.Tasks/Tasks/AssemblyModifierPipeline.cs

```diff
@@ -157,28 +171,47 @@ void RunPipeline (AssemblyPipeline pipeline, ITaskItem source, ITaskItem destina
 			IsUserAssembly = ResolvedUserAssemblies.Any (a => a.ItemSpec == source.ItemSpec),
 		};
 
-		var changed = pipeline.Run (assembly, context);
-
-		if (changed) {
-			Log.LogDebugMessage ($"Saving modified assembly: {destination.ItemSpec}");
-			Directory.CreateDirectory (Path.GetDirectoryName (destination.ItemSpec));
-			writerParameters.WriteSymbols = assembly.MainModule.HasSymbols;
-			assembly.Write (destination.ItemSpec, writerParameters);
-		} else {
-			// If we didn't write a modified file, copy the original to the destination
-			CopyIfChanged (source, destination);
-		}
+		pipeline.Run (assembly, context);
 	}
 
-	void CopyIfChanged (ITaskItem source, ITaskItem destination)
+	class SaveChangedAssemblyStep : IAssemblyModifierPipelineStep
```

Comment:
Any particular reason that this is a nested type?

Reply:
Nope, unnested.